dataland-backend/src/main/kotlin/org/dataland/datalandbackend/validator/DataPointMaximumValue.kt

```diff
@@ -0,0 +1,50 @@
+package org.dataland.datalandbackend.validator
+
+import jakarta.validation.Constraint
+import jakarta.validation.ConstraintValidator
+import jakarta.validation.ConstraintValidatorContext
+import jakarta.validation.Payload
+import org.dataland.datalandbackend.interfaces.datapoints.BaseDataPoint
+import org.dataland.datalandbackendutils.exceptions.InvalidInputApiException
+import java.math.BigDecimal
+import kotlin.properties.Delegates
+import kotlin.reflect.KClass
+
+/**
+ * Annotation for the validation of an ExtendedDataPoint<*> holding a number
+ */
+@Target(AnnotationTarget.FIELD)
+@Constraint(validatedBy = [DataPointMaximumValidator::class])
+annotation class DataPointMaximumValue(
+    val maximumValue: Long = 100,
+    val message: String = "Input validation failed: A base data point holding a number is larger than the set maximum.",
+    val groups: Array<KClass<*>> = [],
+    val payload: Array<KClass<out Payload>> = [],
+)
+
+/**
+ * Class holding the validation logic for a class that implements BaseDataPoint<*> Interface featuring a number
+ */
+class DataPointMaximumValidator : ConstraintValidator<DataPointMaximumValue, BaseDataPoint<*>> {
+    private var maximumValue by Delegates.notNull<Long>()
+
+    override fun initialize(constraintAnnotation: DataPointMaximumValue) {
+        this.maximumValue = constraintAnnotation.maximumValue
+    }
+
+    override fun isValid(dataPoint: BaseDataPoint<*>?, context: ConstraintValidatorContext?): Boolean {
+        return if (dataPoint == null) {
+            true
+        } else {
+            when (dataPoint.value) {
```

Comment:
So is my understanding correct that this function returns true, when the datapoint is null or is of type BigDecimal or Long and the value is smaller than the limit. Otherwise an invalid input exception is thrown. Correct?

Reply:
yes. to be precise, either the datapoint or the value ar allowed to be null

Reply:
ok